zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -1649,4 +1673,31 @@ mod test {
             .block_modifier(Box::new(|block| block.circuits_params.max_txs = 3))
             .run();
     }
+
+    // TODO: add a negative case after upgrading geth.
```

Comment:
TODO needs to be completed

Reply:
Update and add more test cases in commit 7b90336:

- add test cases for equal, less and more sender balance, and cases for max_fee_per_gas and max_priority_fee_per_gas. I move them to `tx_eip1559.rs`.

- fix `max_priority_fee_per_gas <=> gas_tip_cap` and `max_fee_per_gas <=> gas_fee_cap` according to [geth code](https://github.com/ethereum/go-ethereum/blob/master/internal/ethapi/transaction_args.go#L260).

- Fix gas_price of `eth_types::geth_types::Transaction` from U256 to `Option<U256>`, since I want to test the tracing error if `max_fee_per_gas < max_priority_fee_per_gas` for EIP-1559, and they're reset to gas_price if it exists in [geth-utils code](https://github.com/scroll-tech/zkevm-circuits/blob/develop/geth-utils/l1geth/trace.go#L168).